MIGRATION.md

```diff
@@ -1,37 +1,79 @@
+Due to the structural changes of `logr@v1`, several features of this library have been changed to better reflect the structure of the `logr` library. Most notable of these changes include the lack of a singleton logger. It is recommended that developers create an instance a logger with `NewLogger` or `NewLoggerWithOptions` and either keep it as their own singleton or pass the instance throughout the application to use where needed. Methods like `Info`, `Error` are still callable with a `logr.Logger` instance.
 
-- Instead of `log.SetOutput()`, `log.SetLogLevel()`, use either of the following methods:
+ex:
 
-Method 1:
-```go
-l := log.DefaultLogger()
-s, err := log.GetSink(l)
+```golang
+import (
+    "github.com/ViaQ/logerr/v2/log"
+)
 
-if err != nil {
-    // Some action
-}
+var logger := log.NewLogger("")
 
-s.SetVerbosity(1)
-s.SetOutput(ioutil.Discard)
-
-l.Info("hello world")
+logger.Info("Now logging info message")
+logger.Error(errors.New("New error"), "Now logging new error")
+logger.V(1).Info("And logging a second info message")
 ```
 
-Method 2:
-```go
-l := log.DefaultLogger()
-// This method panics, but DefaultLogger will not
-// panic because it uses Sink.
-log.MustGetSink(l).SetVerbosity(1)
-log.MustGetSink(l).SetOutput(ioutil.Discard)
+## Removal of explicit `SetOutput` and `SetLevel`
+
+As a byproduct of the changes in `logr@v1`, these methods have been moved into the internal logging implementation of `logr.Sink`: `Sink`. Changing the internal implementation is considered to be an "advanced" setting. There are two ways to set a logger with a different writer or log level:
+
+```golang
+// Method 1
+
+import (
+    "bytes"
+    "github.com/ViaQ/logerr/v2/log"
+)
+
+buf := bytes.NewBuffer(nil)
+logger := log.NewLogger("", &log.Options{Writer: buf, LogLevel: 1})
+
+logger.Info("I will log at level 1")
+
+// Method 2
+
+import (
+    "bytes"
+    "github.com/ViaQ/logerr/v2/log"
+    "github.com/ViaQ/logerr/v2/internal/sink"
+)
+
+buf := bytes.NewBuffer(nil)
+logger := log.NewLogger("")
 
-l.Info("hello world")
+sink := logger.GetSink().(*sink.Sink)
+sink.SetOutput(buf)
+sink.SetVerbosity(1)
+
+logger.Info("I will log at level 1")
 ```
 
-- Instead of `log.UseLogger`, `log.GetLogger`, keep the logger instance until it is no longer needed
+## `NewLoggerV` vs `V`
+
+`NewLoggerV` is a helper function which calls the logger's `V` method. The difference between the two implementations is that `V` updates the logger's level without updating the `Sink`'s `Verbosity`. This can cause the new logger to not be able to log anything without a manually getting the sink and updating the verbosity. If the logger does not use the `Sink` as a logging interface, the `logr.Logger.V` method is called and returned as normally without any changes done to the internal sink.
+
+Manual Method:
+
+```golang
+import (
+    "bytes"
+    "github.com/ViaQ/logerr/v2/log"
+    "github.com/ViaQ/logerr/v2/internal/sink"
```

Comment:
You can not import an internal package from outside your current "scope" (you can only go "down" in the directory hierarchy but not up or to another module), so this example will not work.

My understanding was that this is the whole reason for the existence of `log.NewLoggerV()`. Isn't this the case?